app/src/main/java/com/nervousfish/nervousfish/data_objects/Contact.java

```diff
@@ -11,15 +11,22 @@
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.List;
+import java.util.Objects;
+
+import nl.tudelft.ewi.ds.bankver.IBAN;
 
 /**
  * A Java object representation (POJO) of a contact.
  */
+@SuppressWarnings({"PMD.ImmutableField", "PMD.UselessParentheses"})
+//1. We don't want the field iban to be final.
+//2. The parentheses on line 125 are not useless.
 public final class Contact implements Serializable {
 
     private static final long serialVersionUID = -4715364587956219157L;
     private final String name;
     private final List<IKey> keys = new ArrayList<>();
+    private IBAN iban;
```

Comment:
If you make this final, then the whole class is immutable again. You can assign null in the 2 remaining constructors as unused value. We want to have all data objects immutable because immutable objects are simple (object can only be in a single state, namely the state when in was created), they are thread-safe (require no synchronization) and they can be shared without cloning (defensive copies), even their internals